### PR TITLE
Fix user home path when running with sudo

### PR DESCRIPTION
**Pull request by snacsnoc**
`util.py` would fail to create `~/.onedrive` when running as root due to incorrect path. `HOME_PATH` would default to `//username` instead of `/home/username`. 

There might be a better way of handling this, rather than hardcoding `home` to `HOME_PATH`.

### CODE REVIEW

File: onedrive_d/config.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 	# when in SUDO, fix the HOME_PATH
 	# may not be necessary on most OSes
 	# buggy!!
-	HOME_PATH = os.path.split(HOME_PATH)[0] + "/" + LOCAL_USER
```
Comment (snacsnoc) [inline onedrive_d/config.py:19]: Again, this would fail due to trying to read `//username` instead of `/home/username`.

**Comment by xybu** [comment]
Hi,

Somehow when I run the code on Xubuntu 14.04 b2, I got my root dir to be "/home/root" instead of "/" (which I didn't pay enough attention until now) and the fix may not work on my environment. Is there a better way to determine the home dir of a particular user (now that we know the actual username)?

**Comment by xybu** [comment]
Will `os.path.expanduser('~' + LOCAL_USER)` or `os.getenv('HOME')` get the actual user's home directory on your system, both root and un-root?

**Comment by snacsnoc** [comment]
I'll do some testing and try it out.

**Comment by snacsnoc** [comment]
`os.getenv('USERPROFILE') or os.getenv('HOME')` works when running the script as non-root. 

One possible work around of this to run `./inst install` without sudo. the `inst` script will still install the required prerequisites with sudo, but fails on line 167 (see https://github.com/snacsnoc/onedrive-d/blob/master/onedrive_d/util.py#L169) when trying to install to `/etc/init.d/`.

**Comment by xybu** [comment]
Stuffing things to /etc/init.d needs sudo permission.

Should I move that script from Python part to `inst` bash code?

Also when the program runs as a daemon, I believe it runs with non-elevated user permission. Need to double check.

**Comment by snacsnoc** [comment]
That's definitely an idea, that way we don't have to deal with figuring out home directories, and just get elevated permissions when necessary.

**Comment by snacsnoc** [comment]
Please see my last commit. Just tested it and it the paths are correct by running `inst` with sudo and non-sudo.

**Comment by xybu** [comment]
I just made a file at https://github.com/xybu92/onedrive-d/blob/wip/experiments/find_realuser/getmypath.py

Could you run it without `sudo`, with `sudo`, and `sudo -s && ./getmypath.py` and see if all the output are the same and correct?

**Comment by xybu** [comment]
Ahh it gets closed after confirmation...

Open a new issue to continue the discussion?

**Comment by snacsnoc** [comment]
With your script, I get the same output by running with and without `sudo`. `sudo -s && ./getmypath.py` simply logs me in as root.

**Comment by xybu** [comment]
```
sudo -s
./getmypath.py
```

?

If this works I feel it more portable than hardwiring string literals in the path.

**Comment by snacsnoc** [comment]
My mistake, I get the same output as the other commands, as in, current non-sudo user.

**Comment by xybu** [comment]
So it will work on your environment if I use this mechanism in config.py and util.py, right?
I have made sure this works on my environment.

**Comment by snacsnoc** [comment]
I'll submit a new pull request with a better method.

**Comment by xybu** [comment]
I'll add you to the co-author list so you can control the merge.

**Pull request merged**